pkg/frontend/admin_vm_actions.go

```diff
@@ -11,43 +11,36 @@ import (
 
 	"github.com/Azure/ARO-RP/pkg/api"
 	"github.com/Azure/ARO-RP/pkg/database/cosmosdb"
+	"github.com/Azure/ARO-RP/pkg/frontend/adminactions"
 )
 
-type azVmActionsWrapper struct {
-	vmName string
-	doc    *api.OpenShiftClusterDocument
-}
-
-func (f *frontend) newAzureActionsWrapper(log *logrus.Entry, ctx context.Context, vmName, resourceID string, vars map[string]string) (azVmActionsWrapper, error) {
-	err := validateAdminVMName(vmName)
+func (f *frontend) prepareAdminActions(log *logrus.Entry, ctx context.Context, vmName, resourceID string, vars map[string]string) (azureActions adminactions.AzureActions, doc *api.OpenShiftClusterDocument, err error) {
+	err = validateAdminVMName(vmName)
 	if err != nil {
-		return azVmActionsWrapper{}, err
+		return
 	}
 	if err != nil {
-		return azVmActionsWrapper{}, err
+		return
 	}
 
-	doc, err := f.dbOpenShiftClusters.Get(ctx, resourceID)
+	doc, err = f.dbOpenShiftClusters.Get(ctx, resourceID)
 	switch {
 	case cosmosdb.IsErrorStatusCode(err, http.StatusNotFound):
-		return azVmActionsWrapper{}, api.NewCloudError(http.StatusNotFound, api.CloudErrorCodeResourceNotFound, "",
+		return nil, doc, api.NewCloudError(http.StatusNotFound, api.CloudErrorCodeResourceNotFound, "",
```

Comment:
nit: if the doc retrieval errors out with 'not found' why return the doc here? Can this be `return nil, nil, api.NewCloudError(...)`?

Reply:
you're right, I actually thought it could be useful to have the doc even if it was not found, but it's `nil` anyways so it's better to be explicit